source/agora/consensus/EnrollmentManager.d

```diff
@@ -170,13 +170,23 @@ public class EnrollmentManager
     ***************************************************************************/
 
     public this (ManagedDatabase stateDB, ManagedDatabase cacheDB,
-                 KeyPair key_pair, immutable(ConsensusParams) params)
+        KeyPair key_pair, immutable(ConsensusParams) params,
+        PreImageCycle cycle = PreImageCycle.init)
```

Comment:
In this case, it would be better to have an overload that just forwards to this.

Reply:
Right. Looks better

Reply:
Ping?

Reply:
I intended to explain later. I tried to change the code like the following but it doesn't work. An error like `Program exited with code -8` occurred.

```
public this (ManagedDatabase stateDB, ManagedDatabase cacheDB,
       KeyPair key_pair, immutable(ConsensusParams) params,
       PreImageCycle cycle)
 {
     this.log = Logger(__MODULE__);
     assert(params !is null);
     this.params = params;
     this.key_pair = key_pair;

     this.cycle = cycle;

     this.db = stateDB;
     this.validator_set = new ValidatorSet(stateDB, params);
     this.enroll_pool = new EnrollmentPool(cacheDB);


     // create the table for enrollment data for a node itself
     this.db.execute("CREATE TABLE IF NOT EXISTS node_enroll_data " ~
         "(key TEXT PRIMARY KEY, val TEXT NOT NULL)");

     // load enrollment key
     this.enroll_key = this.getEnrollmentKey();
 }

 public this (ManagedDatabase stateDB, ManagedDatabase cacheDB,
     KeyPair key_pair, immutable(ConsensusParams) params)
 {
     if (key_pair != KeyPair.init)
         this(stateDB, cacheDB, key_pair, params,
             PreImageCycle(key_pair.secret, params.ValidatorCycle));
     else
         this(stateDB, cacheDB, key_pair, params, PreImageCycle.init);
 }

 /// Unittest-only constructor
 version (unittest) private this (
     KeyPair key_pair, immutable(ConsensusParams) params, PreImageCycle cycle)
 {
     this(new ManagedDatabase(":memory:"), new ManagedDatabase(":memory:"),
         key_pair, params, cycle);
 }

 /// Ditto
 version (unittest) private this (
     KeyPair key_pair, immutable(ConsensusParams) params)
 {
     if (key_pair != KeyPair.init)
         this(key_pair, params, PreImageCycle(key_pair.secret, params.ValidatorCycle));
     else
         this(key_pair, params, PreImageCycle.init);
 }
```

Reply:
We decided to leave the ctor as it is also.